### PR TITLE
DOC: Tidy Utils Docstrings

### PR DESCRIPTION
**Pull request by John-P**
Tidy docstrings in utils:
- Improve line wrapping
- Make indentation consistent
- Add missing tests to improve coverage

### CODE REVIEW

File: tiatoolbox/utils/metrics.py [hunk 1]
```diff
@@ -25,51 +25,57 @@
...
-        unpairedB (ndarray): indices of unpaired points in set A.
+        tuple:
+            - :class:`numpy.ndarray` - Pairing:
+                An array of shape Kx2, each item in K contains indices
+                where point at index [0] in set A paired with point in
+                set B at index [1].
+            - :class:`numpy.ndarray` - Unpaired A:
+                Indices of unpaired points in set A.
+            - :class:`numpy.ndarray` - Unpaired B:
+                Indices of unpaired points in set A.
```
Comment (ghadjigeorghiou) [inline tiatoolbox/utils/metrics.py:56]: Typo: Change 'set A' to 'set B'

File: tiatoolbox/utils/misc.py [hunk 2]
```diff
@@ -722,16 +749,17 @@ def model_to(on_gpu, model):
 def get_bounding_box(img):
     """Get bounding box coordinate information.
 
-    Given an image with zero and non-zero values. This function
-    will return the the minimal box that contains all non-zero
-    values.
+    Given an image with zero and non-zero values. This function will
+    return the the minimal box that contains all non-zero values.
```
Comment (ghadjigeorghiou) [inline tiatoolbox/utils/misc.py:753]: Typo: Remove one of the double 'the'

File: tiatoolbox/utils/misc.py [hunk 3]
```diff
@@ -805,21 +841,28 @@ def prepare_file_dir_cli(img_input, output_path, file_types, mode, sub_dirname):
...
+            One of "wsi" or "tile" mode.
 
     Returns:
-        files_all (list): list of file paths to process.
-        masks_all (list): list of masks corresponding to input files.
-        output_path (pathlib.Path): output path
+        list:
+            List of file paths to process.
+        list:
+            List of masks corresponding to input files. output_path
```
Comment (ghadjigeorghiou) [inline tiatoolbox/utils/misc.py:864]: output_path on the wrong line.

File: tiatoolbox/utils/visualization.py [hunk 4]
```diff
@@ -62,22 +65,31 @@ def overlay_prediction_mask(
...
+        prediction (ndarray):
+            2D prediction map. Multi-class prediction should have values
+            ranging from 0 to N-1, where N is the number of classes.
+        label_info (dict):
+            A dictionary contains the mapping for each integer value
+            within `prediction` to its string and color. [int] : (str,
+            (int, int, int)). By default, integer will be taken as label
+            and color will be random.
+        min_val (float):
+            Only consider predictions greater than or equal to
```
Comment (ghadjigeorghiou) [inline tiatoolbox/utils/visualization.py:79]: I think consider should be considers?

File: tiatoolbox/utils/visualization.py [hunk 5]
```diff
@@ -191,21 +203,31 @@ def overlay_probability_map(
...
+            HW.
+        prediction (ndarray):
+            2D prediction map. Values are expected to be between 0-1.
+        alpha (float):
+            Opacity value used for the overlay.
+        colour_map (string):
+            The colour map to use for the heatmap. `jet` is used as the
+            default.
+        min_val (float):
+            Only consider pixels that are greater than or equal to
```
Comment (ghadjigeorghiou) [inline tiatoolbox/utils/visualization.py:217]: Same as above. Consider -> considers

File: tiatoolbox/utils/visualization.py [hunk 6]
```diff
@@ -351,20 +381,26 @@ def plot_graph(
...
-        edge_size (int): Linewidth of the edge.
+        canvas (np.ndarray):
+            Canvas to be drawn upon.
+        nodes (np.ndarray):
+            List of nodes, expected to be Nx2 where N is the number of
+            nodes. Each node is expected to be of `(x, y)` and should be
+            within the height and width of the canvas.
+        edges (np.ndarray):
+            List of edges, expected to be Mx2 where M is the number of
+            edges. Each edge is defined as `(src, dst)` where each is
```
Comment (ghadjigeorghiou) [inline tiatoolbox/utils/visualization.py:392]: "Each edge is defined as `(src, dst)` where each is respectively the index of within `nodes`."
This doesn't make sense to me. Could we rephrase it?

**Review by ghadjigeorghiou — CHANGES_REQUESTED** [review]

File: tiatoolbox/utils/metrics.py [hunk 7]
```diff
@@ -25,51 +25,57 @@
...
-            of M different points.
-        radius: valid area around a point in setA to consider a given
+        set_a (ndarray):
+            An array of shape Nx2 contains the of XY coordinate of N
+            different points.
+        set_b (ndarray):
+            An array of shape Mx2 contains the of XY coordinate of M
+            different points.
+        radius:
+            Valid area around a point in setA to consider a given
```
Comment (adamshephard) [inline tiatoolbox/utils/metrics.py:44]: set A

File: tiatoolbox/utils/metrics.py [hunk 8]
```diff
@@ -25,51 +25,57 @@
...
             coordinate in setB a candidate for matching.
```
Comment (adamshephard) [inline tiatoolbox/utils/metrics.py:45]: set B

**Comment by John-P** [comment]
Thanks @ghadjigeorghiou, I've review your suggestions and made changes.

File: tiatoolbox/utils/misc.py [hunk 9]
```diff
@@ -761,20 +791,26 @@ def string_to_tuple(in_str):
...
+        mode (str):
+            One of "wsi" or "tile" mode.
+        sub_dirname (str):
+            Name of subdirectory to save output.
 
     Returns:
-        files_all (list): list of file paths to process.
-        output_path (pathlib.Path): updated output path.
+        list:
+            List of file paths to process. output_path (pathlib.Path):
```
Comment (adamshephard) [inline tiatoolbox/utils/misc.py:812]: Separate out the two variables into list and output_path

File: tiatoolbox/utils/misc.py [hunk 10]
```diff
@@ -805,21 +841,28 @@ def prepare_file_dir_cli(img_input, output_path, file_types, mode, sub_dirname):
...
+        file_types (str):
+            File types to process using cli.
+        mode (str):
+            One of "wsi" or "tile" mode.
 
     Returns:
-        files_all (list): list of file paths to process.
-        masks_all (list): list of masks corresponding to input files.
-        output_path (pathlib.Path): output path
+        list:
```
Comment (adamshephard) [inline tiatoolbox/utils/misc.py:861]: give different names to the two output lists?

**Review by adamshephard — CHANGES_REQUESTED** [review]
A few typos to sort.

**Review by ghadjigeorghiou — APPROVED** [review]

File: tiatoolbox/utils/misc.py (hunk 10 above)
Comment (John-P) [reply tiatoolbox/utils/misc.py]: Can you clarify whate you mean here?

Comment (adamshephard) [reply tiatoolbox/utils/misc.py]: Should the two output lists not have different names?

**Review by adamshephard — APPROVED** [review]

**Pull request merged**